### DISCUSSION TITLE
Windows development environment

### DISCUSSION THREAD

**Discussion by mlapaglia** [LycheeOrg/Lychee#1911] [Q&A] [answered]
I'm am looking to contribute to this project. What is the fastest way to spin up a development environment on windows? I already have mysql running in docker and I found a simple php web server for windows https://www.usbwebserver.net/webserver/ .

I am also interested to know how the PRs work across the frontend and backend if the change involves both repos.

Help is appreciated thanks!

**Comment by ildyria** [comment, marked as answer]
Hi, welcome to Lychee.
I would say the best way would be to either use a full Linux distro, or run WSL2 with Linux in it. This will make it significantly easier for you to get set up and fix bugs. Docker is also quite compatible with WSL2 so this should not be a problem either. https://askubuntu.com/a/1397300

With regard to front-end back-end. You make a PR for both repo and in the Front end you just use the same branch name, and add a pointer to the PR on this repo.
The merging process is then as follows:
1. Merge on Front
2. Rebuild front
3. Update the Lychee-front submodule to point master
4. commit on base repo (still your branch).
5. Merge on this repo.

All this is discussed here: https://lycheeorg.github.io/docs/contributions.html :)

We do have some strict coding style and tests requirements. :)
PhpStan will ensure that most of your code is bug free. Note that some functions are forbidden _e.g._ `empty()` or `==`. PhpStan will complain when finding them.

You can also join us on gitter (link the read me) and feel free to ask questions. We have quite a few issues to be worked on, it is pretty much free for all, pick what ever you want to work on.
